File: programming-languages/Python/small_programs/timer.py
import time

from contextlib import contextmanager

@contextmanager
def timer(task_name="Task"):
    start = time.perf_counter()

    try:
        yield
    finally:
        elapsed = time.perf_counter() - start

        hours, remainder = divmod(elapsed, 3600)
        minutes, seconds = divmod(remainder, 60)
        
        print(f"Total Time: {int(hours):02}:{int(minutes):02}:{int(seconds):02}")


with timer('example'):
    time.sleep(2)

File: programming-languages/Rust/small_programs/timer.rs
use std::time::Instant;
use std::thread;
use std::time::Duration;

fn main() {
    let start = Instant::now();

    thread::sleep(Duration::from_secs(2));

    let duration = start.elapsed();
    let total_secs = duration.as_secs();

    let hours = total_secs / 3600;
    let minutes = (total_secs % 3600) / 60;
    let seconds = total_secs % 60;

    println!("Total Time: {:02}:{:02}:{:02}", hours, minutes, seconds);
}
